### PR TITLE
bugfix/range-selector-dropdown-alignment

### LINKED ISSUE #21945: The `rangeSelector.buttonPosition.align` option does not work correctly since v11.3.0.
#### Expected behaviour
Setting [rangeSelector.buttonPosition.align](https://api.highcharts.com/highstock/rangeSelector.buttonPosition.align) to `right` correctly positions the button on the right and maintains responsiveness.

#### Actual behaviour
Setting the [rangeSelector.buttonPosition.align](https://api.highcharts.com/highstock/rangeSelector.buttonPosition.align) option to `right` causes the button to be moved slightly to the right, but incorrectly and does not respond to changing the chart width.

#### Live demo with steps to reproduce
https://jsfiddle.net/BlackLabel/Lwvf0snz/

#### Product version
Doesn't work since v11.3.0, the last working version is v10.2.0.

### PR DESCRIPTION
**Pull request by bm64**
Fixes #21945, a regression since v11.3 causing the `rangeSelector.align` option not to work properly when set to `right`.

### CODE REVIEW

**Review by pawellysy — CHANGES_REQUESTED** [review]
Please invastigate the visual diff

**Review by pawellysy — APPROVED** [review]
Looks good to me! @pawelfus Should we consider adding some tests?

**Review by pawelfus — CHANGES_REQUESTED** [review]
Missing test. I know we have plenty of visual tests, but somehow this regression got through. 

Could you create a generic test, which will check different positions and overlapping? I mean something like all left, all right, combinations.

The code looks good. Visual changes too.

File: ts/Stock/RangeSelector/RangeSelectorDefaults.ts [hunk 1]
```diff
@@ -494,7 +494,7 @@ const rangeSelector: RangeSelectorOptions = {
          * The alignment of the input box. Allowed properties are `left`,
          * `center`, `right`.
          *
-         * @sample {highstock} stock/rangeselector/input-button-position/
+         * @sample {highstock} stock/rangeselector/input-button-opposite-alignment/
```
Comment (pawelfus) [inline ts/Stock/RangeSelector/RangeSelectorDefaults.ts:497]: Use both demos here (and below)

**Review by pawelfus — CHANGES_REQUESTED** [review]
Minor change, see below

**Review by pawelfus — APPROVED** [review]
Thanks!

**Review by kamil-musialowski — APPROVED** [review]
nice!

**Review by DJTechnoo — APPROVED** [review]

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Pull request merged**